### PR TITLE
REGR: groupby with as_index=False on an empty frame

### LINKED ISSUE #41998: REGR: 1.3 behavior change with groupby and asindex=False
#### Code Sample, a copy-pastable example

```python
pd.DataFrame(columns=["date", "crew_member_id"])
    .groupby(["date", "crew_member_id"], as_index=False)
    .sum()
```

#### Problem description

With 1.3rc1 this fails with `ValueError: Length of values (0) does not match length of index (1)` while 1.2 returns and empty frame.

#### Expected Output

If this is the expected behavior fine. Just wanted to ensure that it is not an accidental regression.

### PR DESCRIPTION
**Pull request by rhshadrach**
- [x] closes #41998
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them

No whatsnew because this was introduced in 1.3

### CODE REVIEW

File: pandas/core/groupby/generic.py [hunk 1]
```diff
@@ -1676,7 +1676,8 @@ def _wrap_transformed_output(
 
     def _wrap_agged_manager(self, mgr: Manager2D) -> DataFrame:
         if not self.as_index:
-            index = Index(range(mgr.shape[1]))
+            rows = mgr.shape[1] if mgr.shape[0] > 0 else 0
```
Comment (jbrockmendel) [inline pandas/core/groupby/generic.py:1679]: comment here about why, pointing back to the issue/pr?

**Comment by jbrockmendel** [comment]
This looks right to me, but merits another pair of eyeballs

File: pandas/core/groupby/generic.py (hunk 1 above)
Comment (rhshadrach) [reply pandas/core/groupby/generic.py:1679]: Thanks - added.

Comment (jreback) [reply pandas/core/groupby/generic.py:1680]: we should strive to move this code inside internal (separate PR for sure)

**Comment by jreback** [comment]
this looks good, thanks @rhshadrach

**Comment by jreback** [comment]
@meeseeksdev backport 1.3.x

**Pull request merged**